crates/subspace-core-primitives/Cargo.toml

```diff
@@ -19,6 +19,7 @@ dusk-bls12_381 = { version = "0.11", default-features = false, features = ["allo
 dusk-bytes = "0.1"
 dusk-plonk = { version = "0.12.0", default-features = false, features = ["alloc"], git = "https://github.com/subspace/plonk", rev = "193e68ba3d20f737d730e4b6edc757e4f639e7c3" }
 hex = { version  = "0.4.3", default-features = false }
+libp2p = {version = "0.46.1", optional=true, default-features = false }
```

Comment:
Hm... I think it is really quite undesirable to make `subspace-core-primitives` depend on libp2p :disappointed: 

Also attention to detail:
```suggestion
libp2p = { version = "0.46.1", optional = true, default-features = false }

Reply:
Well, `PieceIndexHash` already has `AsRef<[u8]>` implemented for it. Which means all you need to write is `MultihashDigest::digest(&Code::Identity, piece_index_hash.as_ref())`, which depending on how frequently that is used likely not worth building libp2p, which takes non-negligible amount of time, just to have `From` conversion.

Reply:
Also one whitespace is still missing with latest changes.